packages/labs/react/src/test/use-controller_test.ts

```diff
@@ -69,33 +69,63 @@ suite('useController', () => {
     );
   };
 
-  test('basic lifecycle', () => {
-    let testController!: TestController;
-
-    const TestComponent = ({x}: {x: number}) => {
-      testController = useTest('a');
-      return React.createElement('div', {className: 'foo'}, [
-        `x:${x}, a:${testController.a}`,
-      ]);
-    };
-
-    const render = (props: any) => {
-      ReactDOM.render(React.createElement(TestComponent, props), container);
-    };
-
-    render({x: 1});
-    assert.equal(ctorCallCount, 1);
-    assert.equal(container.innerHTML, `<div class="foo">x:1, a:a</div>`);
-    assert.deepEqual(testController.log, ['connected', 'update', 'updated']);
-    const firstTestController = testController;
+  const lifeCycleTest = ({strict}: {strict: boolean}) => {
+    test(`basic lifecycle${strict ? ' - strict mode' : ''}`, () => {
+      let testController!: TestController;
+
+      let componentRenderLog: string[] = [];
+      let componentLayoutEffectLog: string[] = [];
+
+      const TestComponent = ({x}: {x: number}) => {
+        testController = useTest('a');
+        componentRenderLog = [...testController.log];
+        React.useLayoutEffect(() => {
+          componentLayoutEffectLog = [...testController.log];
```

Comment:
why are we testing this?

Reply:
To validate the state of the controller if the user accesses it in a `useLayoutEffect`.

Reply:
Can you add a comment on what this is doing? Reading it it seems like it may be testing that useLayoutEffect is called in the React component. The key part is that it's checking the log state?